### PR TITLE
Prevents crashes in case of pathological book (BL-10286)

### PR DESCRIPTION
**Pull request by andrew-polk**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360)

### CODE REVIEW

**Review by andrew-polk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @andrew-polk)

___
*[src/components/RowOfCards.tsx, line 82 at r1](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360#-MhG1LwN1F4KP3xF8qbK:-MhG1LwN1F4KP3xF8qbL:b-27rnts) ([raw file](https://github.com/bloombooks/bloomlibrary2/blob/69578345107fd5fc1058aafc3a5fe70ff313353c/src/components/RowOfCards.tsx#L82)):*
> ```XML
>     cardSpecs["by-level"] = useCollectionCardSpec(
>         CollectionCardLayout.iconAndBookCount
>     );
> ```

Not actually related to this change, but was just investigating this for another issue and thought I might as well fix it.

File: src/components/RowOfCards.tsx [hunk 1]
```diff
@@ -74,6 +74,12 @@ const RowOfCollectionCards: React.FunctionComponent<{
     cardSpecs["by-topic"] = useCollectionCardSpec(
         CollectionCardLayout.iconAndBookCount
     );
+    // Unexpected. But prevents a crash if it happens.
```
Comment (hatton) [inline src/components/RowOfCards.tsx:77]: Not as sure about this one, and it's a totally different topic... can you split it into its own commit?

**Review by andrew-polk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @hatton)

___
*[src/components/RowOfCards.tsx, line 77 at r1](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360#-MhFz0YTX1p0GvWdlgZH-r1-77:-MhGKRnGB1dCNWx8-0DC:b-896fix) ([raw file](https://github.com/bloombooks/bloomlibrary2/blob/69578345107fd5fc1058aafc3a5fe70ff313353c/src/components/RowOfCards.tsx#L77)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Not as sure about this one, and it's a totally different topic... can you split it into its own commit?
</blockquote></details>

Done.

**Review by andrew-polk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @hatton)

___
*[src/components/RowOfCards.tsx, line 77 at r1](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360#-MhFz0YTX1p0GvWdlgZH-r1-77:-MhGLMlp7PqMR6bKUgqC:byu69jp) ([raw file](https://github.com/bloombooks/bloomlibrary2/blob/69578345107fd5fc1058aafc3a5fe70ff313353c/src/components/RowOfCards.tsx#L77)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Created https://github.com/BloomBooks/BloomLibrary2/pull/362

**Review by hatton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @andrew-polk and @hatton)

___
*[src/components/BookDetail/BookDetailHeaderGroup.tsx, line 152 at r2](https://reviewable.io/reviews/bloombooks/bloomlibrary2/360#-MhLBaak6dJxwOn_RXD9:-MhLBaak6dJxwOn_RXDA:bcamlnd) ([raw file](https://github.com/bloombooks/bloomlibrary2/blob/269c714988491120038ac15647232de7c18f282b/src/components/BookDetail/BookDetailHeaderGroup.tsx#L152)):*
> ```XML
>                     </p> */}
>                         {(props.book.languages &&
>                             props.book.languages.length && (
> ```

This is fine but note that it can be written as `props.book.languages?length`
